vmp/resources/waf/custom_rule_set.go

```diff
@@ -0,0 +1,615 @@
+															"is_regex": {
+																Optional: true,
+																Type:     schema.TypeBool,
+																Description: "Determines whether the value property will be interpreted as a regular expression. \\\n" +
+																	"    Valid values are: \\\n" +
+																	"    *true*: Regular expression \\\n" +
+																	"    *false*: Default value. Literal value.",
+															},
+															"value": {
+																Type:     schema.TypeString,
+																Optional: true,
+																Description: "Restricts the match condition defined by the type property to the specified value. \\\n" +
+																	"Example: \\\n" +
+																	"If the type property is set to REQUEST_HEADERS and this property is set to User-Agent, then this match condition is restricted to the User-Agent request header. If the value property is omitted, then this match condition applies to all request headers.",
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+							Description: "sec_rule object describes a custom rule",
+						},
+					},
+				},
+				Description: "Contains custom rules. Each directive object defines a custom rule via the sec_rule object \\\n" +
+					"    Note: You may create up to 10 custom rules.",
+			},
+		},
+	}
+}
+
+func ResourceCustomRuleSetCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	var diags diag.Diagnostics
+
+	config := m.(**api.ClientConfig)
+
+	wafService, err := buildWAFService(**config)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	accountNumber := d.Get("account_number").(string)
+
+	log.Printf("[INFO] Creating WAF Rate Rule for Account >> %s", accountNumber)
+
+	customRuleSet := sdkwaf.CustomRule{
+		Name: d.Get("name").(string),
+	}
+
+	directive, err := ConvertInterfaceToDirectives(d.Get("directive"))
+	if err != nil {
+		return diag.Errorf("error parsing directive: %+v", err)
+	}
+
+	customRuleSet.Directives = *directive
+
+	log.Printf("[DEBUG] Name: %+v\n", customRuleSet.Name)
+	log.Printf("[DEBUG] Directive(s): %+v\n", customRuleSet.Directives)
+
+	resp, err := wafService.AddCustomRuleSet(customRuleSet, accountNumber)
+
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] %+v", resp)
+
+	d.SetId(resp.ID)
+
+	return diags
+}
+
+func ResourceCustomRuleSetRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	return diags
+}
+
+func ResourceCustomRuleSetUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	return diags
+}
+
+func ResourceCustomRuleSetDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	return diags
+}
+
+func ConvertInterfaceToDirectives(attr interface{}) (*[]sdkwaf.Directive, error) {
```

Comment:
I forgot to mention during our discussion, but we will be using "Flatten" and "Expand" for method names.
API Model = > Terraform Model is "Flatten"
Terraform Model => API Model is "Expand"
So this method should be `ExpandDirectives`

Reply:
Updated.